### PR TITLE
[READY] Nukes archeology component and icon_plating and environment_type in favor of base_icon_state. Dig behaviour code improvement

### LINKED ISSUE #25758: Beach Dome: The Blackening
image

I guess natural tunnels fuck up the generation

### PR DESCRIPTION
**Pull request by Azarak**
## About The Pull Request
Plating updates, including digging from its /asteroid subtype now care about the non volatile base_icon_state.
This fixes bugs regarding random black plating subtypes, most notably the sand in the beach biodome.
Moved the digging related variables into the asteroid level

fixes: https://github.com/tgstation/tgstation/issues/25758

## Why It's Good For The Game
Bugs and abundant variables bad

## Changelog
:cl:
fix: Beach Biodome (and more) will now properly have non-black sand and water
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
This should be fixed properly

**Comment by Azarak** [comment]
> 
> 
> This should be fixed properly instead of just papering over the issue

I can't figure it out and it's frustrating, and the black turfs are bothering me. I dont know turf code well either

**Comment by Azarak** [comment]
The issue only seems to be happening on turfs which get created before SSatoms is initialized

**Comment by LemonInTheDark** [comment]
It happens when you manually spawn the turfs in during the round too

**Comment by LemonInTheDark** [comment]
GOD I'M GOING TO BREAK SOMETHING this is so fucking convoluted

**Comment by LemonInTheDark** [comment]
Unless the issue is somehow being introduced by byond versions? it's older then https://github.com/tgstation/tgstation/pull/49165. Unless it's random at mapload, which I highly doubt, it's fixed 500 commits past 49165, I'm in the process of isolating the commit that broke it, expect more details in around 24 hours.

**Comment by Azarak** [comment]
> 
> 
> It happens when you manually spawn the turfs in during the round too

Huh? Interesting.. I remember when tinkering with map generation during SSmapping, it'd do black turfs, but when having it done after the game starts, it'd work fine.

**Comment by Cyberboss** [comment]
Fixes #25758 ?

**Comment by LemonInTheDark** [comment]
@Azarak this code gives me hives, if you can purge it cleanly please do so

**Comment by Azarak** [comment]
@LemonInTheDark i'll try and purge that icon_plating variable in a bit

**Comment by AnturK** [comment]
Why WIP ?

**Comment by Azarak** [comment]
> 
> 
> Why WIP ?

I'll be removing the icon_plating variable and handling archeology differently. Speaking of it, let me get on that right now, i was kinda busy so the PR sat for some time

**Comment by Azarak** [comment]
Tested spawns and interactions with damaging/repairing plating

File: code/game/turfs/open/floor/plating.dm [hunk 1]
```diff
@@ -35,16 +36,10 @@
 	if (!burnt_states)
 		burnt_states = list("panelscorched")
 	. = ..()
-	if(!attachment_holes || (!broken && !burnt))
-		icon_plating = icon_state
-	else
-		icon_plating = initial(icon_state)
 
 /turf/open/floor/plating/update_icon()
```
Comment (AnturK) [inline code/game/turfs/open/floor/plating.dm:40]: Might as well remove this override ?

Comment (Azarak) [reply code/game/turfs/open/floor/plating.dm:40]: Good catch!

**Comment by AnturK** [comment]
There's probably some edge cases that relied on that update icon change but might be hard to find.

**Pull request merged**